do_kafka_test.sh

```diff
@@ -1,71 +1,172 @@
 #!/bin/bash
-./start_kafka.sh
 
-# Wait until Kafka creates our topics.
-while true; do
-    ntopics=$(docker exec jpocvdp_kafka_1 /opt/kafka/bin/kafka-topics.sh --list --zookeeper 172.17.0.1 | wc -l)
+# This script tests the PPM against a kafka cluster. It sets up variables for container and input data
+# file names. It starts a Kafka container using another script and checks that required topics are created.
+# If the container or topics are missing, the script exits. It builds a Docker image using the current
+# directory and specified name/tag. It runs a series of tests using a script with different properties
+# and input data files, outputting results to the console. It stops the Kafka container after the tests
+# are completed. The script performs five steps: set up the test environment, wait for Kafka to create
+# topics, build the PPM image, run the tests, and clean up.
 
-    if [[ $ntopics == "4" ]]; then 
-        echo 'Found 4 topics:'
-        docker exec jpocvdp_kafka_1 /opt/kafka/bin/kafka-topics.sh --list --zookeeper 172.17.0.1 2> /dev/null
-        
-        break   
-    fi
-
-    sleep 1
-done
+CYAN='\033[0;36m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
 
+CURRENT_DIR_NAME=${PWD##*/}
```

Comment:
Test Scenario 1:
I started Zookeeper and Kafka container first, then run this do_kafka_test.sh script. This script is going to start kafka again in the setup() func, and waitForKafkaToCreateTopics() complain that Kafka is not up. 
![image](https://github.com/usdot-jpo-ode/jpo-cvdp/assets/62157949/f84e7865-0df3-45ae-bf23-1da9b109d1d5)


Test Scenario 2:
Stop Kafka and zookeeper. Run the do_kafka_test.sh script.
![image](https://github.com/usdot-jpo-ode/jpo-cvdp/assets/62157949/33c7d87e-6788-44a6-a380-699ba1294f97)

IT will start the containers and then stop them at step 2. Should we remove the stop_kafka.sh from the start_kafka.sh script?

Reply:
If Zookeeper and Kafka are already running, the `do_kafka_test.sh` script should restart them prior to building the PPM image and running the tests.

If Zookeeper and Kafka are not running, the `do_kafka_test.sh` script should start them.

Another comment of yours mentions that the kafka container name uses an underscore, not a hyphen. It's possible that this is why the container is not being found in Step 2. Interestingly, when I run the script the container name uses a hyphen, not an underscore. I wonder if this could be due to differences in docker versions. I am using Docker Desktop 4.17.0, what are you using?

Reply:
I am using ubuntu 18.04 and docker version 20.10.6